src/main/java/org/hpccsystems/jdbcdriver/HPCCPreparedStatement.java

```diff
@@ -123,6 +134,60 @@ public ResultSet executeQuery() throws SQLException
         return result;
     }
 
+    public static String primitiveToString(Object x) throws IOException
+    {
+        if (x != null)
+        {
+            if (x instanceof String)
+                return (String)x;
+            else if (x instanceof Boolean)
+                return String.valueOf(x);
+            else if (x instanceof Integer)
+                return String.valueOf(x);
+            else if (x instanceof Long)
+                return String.valueOf(x);
+            else if (x instanceof Float)
+                return String.valueOf(x);
+            else if (x instanceof Byte)
+                return Byte.toString((Byte)x);
+            else if (x instanceof Short)
+                return Short.toString((Short)x);
+            else if (x instanceof Double)
+                return String.valueOf(x);
+            else if (x instanceof BigDecimal)
+                return ((BigDecimal)x).toString();
+            else if (x instanceof byte[])
+                return new String((byte[])x, StandardCharsets.UTF_8);
+            else if (x instanceof Time)
+                return ((Time)x).toString();
+            else if (x instanceof java.sql.Date)
+                return ((java.sql.Date)x).toString();
+            else if (x instanceof Timestamp)
+                return ((Timestamp)x).toString();
+            else if (x instanceof InputStream)
+            {
+                ByteArrayOutputStream result = new ByteArrayOutputStream();
+                byte[] buffer = new byte[1024];
+                int length;
+                while ((length = ((InputStream)x).read(buffer)) != -1)
```

Comment:
Can you just read until you read less than 1024 bytes?  It looks like you will call read again after reading less than 1024 bytes when there is nothing more to read.